source/1_engine/rex_engine/src/diagnostics/logging/logger.cpp

```diff
@@ -1,71 +1,77 @@
 #include "rex_engine/diagnostics/logging/logger.h"
 
-#include <rex_engine/diagnostics/logging/internal/sinks/basic_file_sink.h>
-#include <rex_engine/diagnostics/logging/internal/sinks/stdout_color_sinks.h>
-#include <rex_std/filesystem.h>
-#include <rex_std/unordered_map.h>
-#include <rex_std/utility.h>
-#include <rex_std/vector.h>
+#include "rex_engine/diagnostics/logging/internal/sinks/basic_file_sink.h"
+#include "rex_engine/diagnostics/logging/internal/sinks/stdout_color_sinks.h"
+#include "rex_engine/memory/debug_allocator.h"
+#include "rex_engine/memory/global_allocator.h"
+#include "rex_engine/memory/untracked_allocator.h"
+#include "rex_std/filesystem.h"
+#include "rex_std/unordered_map.h"
+#include "rex_std/utility.h"
+#include "rex_std/vector.h"
 
 namespace rex
 {
   using LogPattern  = const char*;
-  using LogLevelMap = rsl::unordered_map<LogVerbosity, rexlog::level::level_enum>;
+  using LogLevelMap = DebugHashTable<LogVerbosity, rexlog::level::LevelEnum>;
 
-  using LoggerObjectPtr    = rsl::shared_ptr<rexlog::logger>;
-  using LoggerObjectPtrMap = rsl::unordered_map<LogCategoryName, LoggerObjectPtr>;
+  using LoggerObjectPtr    = rsl::shared_ptr<rexlog::Logger>;
+  using LoggerObjectPtrMap = DebugHashTable<LogCategoryName, LoggerObjectPtr>;
 
-  LoggerObjectPtrMap g_loggers;
+  LoggerObjectPtrMap& loggers()
+  {
+    static LoggerObjectPtrMap loggers(rex::global_debug_allocator());
+    return loggers;
+  }
 
   //-------------------------------------------------------------------------
   LogLevelMap get_log_levels()
   {
-    return {{LogVerbosity::NoLogging, rexlog::level::off}, {LogVerbosity::Fatal, rexlog::level::critical}, {LogVerbosity::Error, rexlog::level::err},        {LogVerbosity::Warning, rexlog::level::warn},
-            {LogVerbosity::Log, rexlog::level::info},      {LogVerbosity::Verbose, rexlog::level::debug},  {LogVerbosity::VeryVerbose, rexlog::level::trace}};
+    return {{LogVerbosity::NoLogging, rexlog::level::Off}, {LogVerbosity::Fatal, rexlog::level::Critical}, {LogVerbosity::Error, rexlog::level::Err},        {LogVerbosity::Warning, rexlog::level::Warn},
+            {LogVerbosity::Log, rexlog::level::Info},      {LogVerbosity::Verbose, rexlog::level::Debug},  {LogVerbosity::VeryVerbose, rexlog::level::Trace}};
```

Comment:
I forgot how this is done, but since you already have a review comment I can add this simple one, but would you be able to tell the formatter to stop putting these one the same line. 
IMO it is a lot more readable when each entry is on a separate line